vsphere/datadog_checks/vsphere/vsphere.py

```diff
@@ -922,26 +905,25 @@ def check(self, instance):
 
             # Second part: do the job
             self.collect_metrics(instance)
-        else:
-            self.log.debug("Thread pool is still processing jobs from previous run. Not scheduling anything.")
 
-        self._query_event(instance)
+            self._query_event(instance)
 
-        thread_crashed = False
-        try:
-            while True:
-                self.log.error(self.exceptionq.get_nowait())
-                thread_crashed = True
-        except Empty:
-            pass
+            try:
+                thread_crashed = False
+                while True:
+                    # Pop every exceptions from the queue and log them
+                    self.log.error(self.exceptionq.get_nowait())
+                    thread_crashed = True
+            except Empty:
+                pass
 
-        if thread_crashed:
-            self.stop_pool()
-            raise Exception("One thread in the pool crashed, check the logs")
+            if thread_crashed:
+                raise Exception("One thread in the pool crashed, check the logs")
```

Comment:
now, when a thread crashes, the collection over all the threads is terminated. Could this change in behavior make the check less resilient?

Reply:
the behavior is the same as before. All threads were terminated.

Reply:
right, must've misunderstood the code since `stop_pool` used to terminate the pool 👍